src/main/java/kubatech/tileentity/gregtech/multiblock/GT_MetaTileEntity_MegaIndustrialApiary.java

```diff
@@ -472,6 +493,303 @@ public ITexture[] getTexture(
+                            .setOnClick((clickData, widget) -> {
+                                if (!(player instanceof EntityPlayerMP)) return;
+                                if (mStorage.size() <= ID) return;
+                                if (this.mMaxProgresstime > 0) {
+                                    GT_Utility.sendChatToPlayer(player, "Can't eject while running !");
+                                    return;
+                                }
+                                BeeSimulator removed = mStorage.remove(ID);
+                                addOutput(removed.queenStack);
+                                GT_Utility.sendChatToPlayer(player, "Queen ejected !");
+                            })
+                            .setBackground(() -> new IDrawable[] {
+                                getBaseMetaTileEntity().getGUITextureSet().getItemSlot(),
+                                GT_UITextures.OVERLAY_SLOT_BEE_QUEEN,
+                                new ItemDrawable(drawables.size() > ID ? drawables.get(ID) : null)
+                                        .withFixedSize(16, 16, 1, 1)
+                            })
+                            .dynamicTooltip(() -> {
+                                if (drawables.size() > ID)
+                                    return Arrays.asList(
+                                            drawables.get(ID).getDisplayName(),
+                                            EnumChatFormatting.GRAY + "Left click to eject");
+                                return Collections.emptyList();
+                            })
+                            .setSize(18, 18));
+                }
+                beesContainer.widget(row.setPos(0, i * 18).setEnabled(widget -> {
+                    int y = widget.getPos().y;
+                    int cy = beesContainer.getVerticalScrollOffset();
+                    int ch = beesContainer.getVisibleHeight();
+                    return y >= cy - ch && y <= cy + ch;
+                }));
+            }
+        beesContainer.attachSyncer(
+                new FakeSyncWidget.ListSyncer<>(
+                        () -> mStorage.stream().map(s -> s.queenStack).collect(Collectors.toList()),
+                        l -> {
+                            drawables.clear();
+                            drawables.addAll(l);
+                        },
+                        (buffer, i) -> {
+                            try {
+                                buffer.writeItemStackToBuffer(i);
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
+                            }
+                        },
+                        buffer -> {
+                            try {
+                                return buffer.readItemStackFromBuffer();
+                            } catch (IOException e) {
+                                throw new RuntimeException(e);
```

Comment:
I don't think failure to show bees is fatal, print log instead.

Reply:
I will leave this as exception, because that should never happen (each slot must have input itemstack)